### PR TITLE
Updated sftp backend to to fix issue where some servers return a gene…

### LINKED ISSUE #158: Opening RW on read causes issues with some SFTP servers.
**Describe the bug**
A recent update to change Read() calls to open a file with os.O_RDWR returns a generic SSH_FX_Failure error on some servers.

Instead of of opening RW, we should open RO on initial Read and reopen in RW is Write is later called.  This is a bit of a contrived scenario (writing after reading) but it is possible.

### PR DESCRIPTION
**Pull request by funkyshu**
…ric error message when a file is opened for RW on Read(). Fixes #158.

### CODE REVIEW

File: backend/sftp/file.go [hunk 1]
```diff
@@ -281,7 +282,7 @@ func (f *File) Seek(offset int64, whence int) (int64, error) {
 	f.fileSystem.connTimerStop()
 	defer f.fileSystem.connTimerStart()
 
-	sftpfile, err := f.openFile(os.O_RDWR)
+	sftpfile, err := f.openFile(os.O_RDONLY)
```
Comment (dustin-schnee) [inline backend/sftp/file.go:285]: I should have asked this last time, but would you explain what calling openFile 2 times accomplishes?

Comment (dustin-schnee) [inline backend/sftp/file.go:285]: I should have asked this last time, but would you explain what calling openFile 2 times accomplishes.

Comment (funkyshu) [reply backend/sftp/file.go:285]: At this line? What do you mean?

Comment (dustin-schnee) [reply backend/sftp/file.go:285]: Nevermind, ignore me, I failed to look at it closely :)

**Review by dustin-schnee — APPROVED** [review]

**Pull request merged**